Commendingness

TheVanderlin opened the pull request:
<!-- Write **BELOW** The Headers and **ABOVE** The comments else it may not be viewable. -->
<!-- Note: PRs including balance changes authored by anyone other than maintainers and official devs will not be considered. -->

## About The Pull Request
A very simple PR that enables you to commend other characters while playing the round. Ensuring that player characters who start with the letter A aren't defaulted to be commended. I've tested this by playing characters with A, C and M as the initial characters -- Marius never got any commends. Cirilla hardly got any. But Anna/Avaricius have consistently gotten far more PQ then the majority of players I've talked to about this issue. So very clearly the PQ system due to it being round-end and being a pop-up that's forced on you, is causing the majority of players to just randomly select any name starting with A and closing.

I am personally guilty of this, I do it, because honestly I don't remember everyone and it's annoying to see a pop-up you instinctively want to close it.

<!-- Describe The Pull Request. Please be sure every change is documented or this can delay review or prevent the PR from being merged! -->

## Why It's Good For The Game

This fixes the issue by allowing players to commend characters WHILE they are playing. It makes it a more conscious decision, it's not a split second thing. Players will notice someone doing good roleplay, and on the spot spend their singular commend on them. Meaning the best roleplayers get the most commends now -- not people named A'something something.

It's a really big criticism of the game right now, a major complaint -- you read OOC. You guys see the Discord, it's what puts a lot of people off the game and it's also why a lot of new players have a hard time getting into Roguetown.

<!-- Please add a short description of why you think these changes would benefit the game. If you can't justify it in words, it might not be worth adding. -->

## Pre-Merge Checklist
<!-- Don't bother filling these in while creating your Pull Request, just click the checkboxes after the Pull Request is opened and you are redirected to the page. -->
- [ ] You tested this on a local server.
- [ ] This code did not runtime during testing.
- [ ] You documented all of your changes.
<!-- Neither the compiler nor workflow checks are perfect at detecting runtimes and errors. It is important to test your code/feature/fix locally. -->

TheVanderlin commented:
Yes also I hosted locally, and I adjusted the code temporarily to let me commend myself to ensure it's working properly.

It worked, I wasn't able to do it a second time -- and also obviously this PR doesn't include my edit to let me test commends on myself.

TheVanderlin commented:
I've heard there's an alternative.

I would like to broach my own. If you guys don't end up with an alternative yourself.

You could just use the OG "Inspection" commendation where you can commend characters by inspecting their character. Then under the appearance text you could select "Commend"

dwasint commented:
> I've heard there's an alternative.
> 
> I would like to broach my own. If you guys don't end up with an alternative yourself.
> 
> You could just use the OG "Inspection" commendation where you can commend characters by inspecting their character. Then under the appearance text you could select "Commend". I can update this PR with that and remove the original if it's prefered.

I was going to comment asking for this version instead, adding it via the examine info is better

TheVanderlin commented:
![image](https://github.com/user-attachments/assets/b3b97c0f-b995-43eb-a489-659d68b20e59)

TheVanderlin commented:
Okay gonna push the alt, which removes the previous code.

Obviously yes in the example I am commending myself -- that's temp code not actually part of the PR for testing. You can only commend other characters which I've also tested, not letting me commend myself when I have the correct client check for commended = true

TheVanderlin commented:
Ready for Test-Merge.

Sadboysuss commented:
Closes https://github.com/Vanderlin-Tales-Of-Wine/Vanderlin/issues/841

Box2444 commented:
seems like the examine text is put in a place where it'd be pretty easy to misclick

TheVanderlin commented:
> Putting it in examine is really odd. It's meant to be for IC-pertinent information.

You cannot make it anything else without it revealing the characters name which is meta

TheVanderlin commented:
I've added a fire to the HREF issue

TheVanderlin commented:
> > I've added a fire to the HREF issue
> 
> This did not really address the issue... you need to check in the Topic call. You can add protections in the UI, but nothing is stopping clients from just making a BYOND link on their browser and opening it to make a Topic call.

I have moved it to Topic.dm
